discovery-provider/src/tasks/index_tastemaker_notifications.py

```diff
@@ -59,6 +68,33 @@ def dedupe_notifications_by_group_id(repost_notifications, save_notifications):
     return deduped_notifications_by_group_id
 
 
+def dedupe_notifications_from_existing_notifications(
+    tastemaker_notifications_to_add, session
+):
+    filter_expr = and_(
```

Comment:
i think this filter might not be accurate because it's checking if this specifier exists in notifications at all and if the group id exists at all not necessary grouped together, right?

i think it should be a series of ORs with a single group_id AND specifier, right?